src/main/scala/mesosphere/marathon/api/akkahttp/EntityMarshallers.scala

```diff
@@ -95,10 +96,49 @@ object EntityMarshallers {
     jsonStringMarshaller.compose(printer).compose(internal => writes.writes(ramlWrites.write(internal)))
 
   import mesosphere.marathon.raml.AppConversion.appRamlReader
-  implicit def appUnmarshaller(
+  implicit def appDefinitionUnmarshaller(
     implicit
-    normalization: Normalization[raml.App], validator: Validator[AppDefinition]) = {
-    validEntityRaml(playJsonUnmarshaller[raml.App])
+    normalization: Normalization[raml.App],
+    validator: Validator[AppDefinition]): FromEntityUnmarshaller[AppDefinition] = {
+    validEntityRaml(playJsonUnmarshaller[raml.App]).handleValidationErrors
+  }
+
+  implicit val appDefinitionMarshaller: ToEntityMarshaller[AppDefinition] =
```

Comment:
If I'm not mistaken this could just be

```scala
implicit val appDefinitionMarshaller = playJsonMarshaller[appDefinitionMarshaller]
```
or what did I miss?